src/common/snippets/src/utils/utils.cpp

```diff
@@ -316,6 +316,40 @@ int64_t get_dim_stride(const lowered::ExpressionPort& expr_port, size_t idx) {
     return get_stride(dim_idx, expr_port.get_descriptor_ptr()->get_shape());
 }
 
+void visit_path(const lowered::ExpressionPtr& expr,
+                std::unordered_set<lowered::ExpressionPtr>& visited,
```

Comment:
I made `visited` an argument exactly because in my use cases I pass non-empty set. For example, [here](https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/src/runtime_configurator.cpp#L515) we go down the graph from parameters and collect loops to split. In order to avoid double traversal, the single `visited` is used for all parameters